examples/vision/mixUp.py

```diff
@@ -0,0 +1,219 @@
+"""
+Title: mixup augmentation for image classification
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2020/03/06
+Last modified: 2020/03/06
+Description: How to implement mixup to augment images for image classification problems.
+"""
+"""
+## Introduction
+"""
+
+"""
+mixup is a *domain-agnostic* data augmentation technique proposed in [mixup: Beyond
+Empirical Risk Minimization](https://arxiv.org/abs/1710.09412) by Zhang et al. It's
+implemented as the following -
+
+* $\tilde{x}=\lambda x_{i}+(1-\lambda) x_{j}$, where $x_{i}$ and $x_{j}$ are input
+features
+* $\bar{y}=\lambda y_{i}+(1-\lambda) y_{j}$, where $y_{i}$ and $y_{j}$ are one-hot
+encoded labels
+
+The technique is quite holistically named - we are literally mixing up the features and
+their corresponding labels. Implementation-wise it's simple. As you can imagine this
+recipe can be extended to a variety of data modalities such as computer vision, natural
+language processing, speech, and so on.
+
+This example requires TensorFlow 2.4 or higher, as well as TensorFlow Probability, which
+can be installed using the following command:
+"""
+
+"""shell
+pip install tensorflow-probability
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import tensorflow.keras.layers as L
+from tensorflow.keras.utils import to_categorical
+
+import tensorflow_probability as tfp
+
+tfd = tfp.distributions
+
+"""
+## Prepare dataset
+
+We will be using the FashionMNIST dataset for this toy example. But this same recipe can
+be used for other classification datasets as well.
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.fashion_mnist.load_data()
+
+x_train = x_train.astype("float32") / 255.0
+x_train = np.reshape(x_train, (-1, 28, 28, 1))
+y_train = to_categorical(y_train, 10)
+
+x_test = x_test.astype("float32") / 255.0
+x_test = np.reshape(x_test, (-1, 28, 28, 1))
+y_test = to_categorical(y_test, 10)
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 64
+EPOCHS = 5
+
+"""
+## Wrap as TensorFlow datasets
+"""
+
+train_ds_one = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+train_ds_two = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+# As we will be mixing the images and their labels up, we
+# are combining two differently shuffled but SAME datasets.
```

Comment:
Suggestion: refactor

```suggestion
# Because we will be mixing up the images and their corresponding labels, we
# will be combining two shuffled datasets from the same training data.
```